apps/abstraxion-dashboard/hooks/useAbstraxionSigningClient.ts

```diff
@@ -18,41 +22,68 @@ export const useAbstraxionSigningClient = () => {
 
   const stytch = useStytch();
   const sessionToken = stytch.session.getTokens()?.session_token;
-  const { data: grazClient } = useCosmWasmSigningClient();
+
+  const { data } = useOfflineSigners();
+  const keplr = getKeplr();
 
   const [abstractClient, setAbstractClient] = useState<AAClient | undefined>(
     undefined,
   );
 
   useEffect(() => {
-    async function getStytchSigner() {
-      const jwtSigner = new AbstractAccountJWTSigner(
-        abstractAccount.bech32Address,
-        sessionToken,
-      );
+    async function getSigner() {
+      let signer: AbstractAccountJWTSigner | AADirectSigner | undefined =
+        undefined;
+
+      // TODO: authenticator vs index. which one is better long term
+      // How do you get authenticator index if there are duplicates...?
+      // Client-side disable ability to add duplicate authenticators
+      switch (connectionType) {
+        case "stytch":
+          signer = new AbstractAccountJWTSigner(
+            abstractAccount.id,
+            abstractAccount.currentAuthenticatorIndex,
+            sessionToken,
+          );
+          break;
+        case "graz":
+          if (data && data.offlineSigner) {
+            signer = new AADirectSigner(
+              data?.offlineSigner as any, // Temp solution. graz vs internal cosmjs version mismatch
+              abstractAccount.id,
+              abstractAccount.currentAuthenticatorIndex,
+              // @ts-ignore - signArbitrary function exists on Keplr although it doesn't show
+              keplr.signArbitrary,
+            );
+            break;
+          }
+        case "none":
+          // TODO: What do we want to do here?
+          signer = undefined;
+          break;
+      }
 
-      const jwtClient = await AAClient.connectWithSigner(
+      if (!signer) {
+        // TODO: More robust edge handling
```

Comment:
anything we can do here? at least log?

Reply:
Yea we can log, the worst case is that the transactions just fail